backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,346 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// New Backend
+func New() backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "(Required) The address of the REST endpoint.",
+			},
+
+			"update_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "POST",
+				Description: "(Optional) HTTP method to use when updating state. Defaults to POST.",
+			},
+
+			"lock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the lock REST endpoint. Defaults to address.",
+			},
+
+			"lock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "LOCK",
+				Description: "(Optional) The HTTP method to use when locking. Defaults to LOCK.",
+			},
+
+			"unlock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the unlock REST endpoint. Defaults to address.",
+			},
+
+			"unlock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "UNLOCK",
+				Description: "(Optional) The HTTP method to use when unlocking. Defaults to UNLOCK.",
+			},
+
+			"username": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The username for HTTP basic authentication.",
+			},
+
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The password for HTTP basic authentication.",
+			},
+
+			"skip_cert_verification": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "(Optional) Whether to skip TLS verification. Defaults to false.",
+			},
+
+			"local_cert_ca_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "CA to use when the rest api is using a self signed certificate.",
+			},
+
+			"mutual_tls_authentication": {
```

Comment:
I am curious if we can remove this argument entirely. Can't you infer that they want to use mutual TLS auth if `local_cert_file` is set? Likewise, you can infer that it is not wanted if `local_cert_ca_file` is set? If both are set, error?

If only one of `local_cert_file` or `local_cert_ca_file` can be set, you can use `ConflictsWith` in the schema:

https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L166

`ConflictsWith` ensures that a set of keys cannot all be set in the config. So `local_cert_file` conflicts with `local_ca_cert_file`, and `local_ca_cert_file` conflicts with `local_cert_file`. That should also clean up your code a bit, you will know that only one is set when you entire your configure function.